### PR TITLE
Refactored Test cases from TestUtils to Enzyme 

### PR DESCRIPTION
**Pull request by RamuKammagani-osi**
Cleaned up all warnings

### CODE REVIEW

**Comment by edgibbs** [comment]
Can one of the admins verify this patch?

**Comment by mlipardo** [comment]
Resolved the conflicts with GlobalHeader, karmaConf, PageHeader with @NagaPradeepGajula-osi.

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
retest this please

**Comment by NagaPradeepGajula-osi** [comment]
@edgibbs can you test it again. I believe I fixed it.

**Comment by edgibbs** [comment]
retest this please

**Comment by RamuKammagani-osi** [comment]
I tested it... its passing !

File: src/test/InputComponent_test.js [hunk 1]
```diff
@@ -1,17 +1,80 @@
...
+    placeholder: 'string ',
+    value: 'enter the name',
+  };
+  const inst = wrapper.instance();
+  wrapper.setProps(input);
 
-  it('renders the tag', function() {
-    expect(resultTag.type).toBe('div');
+  it('verify the no of "div" tags ', () => {
+    expect(wrapper.find('div').length).toEqual(2);
```
Comment (edgibbs) [inline src/test/InputComponent_test.js]: Could probably delete this one.

File: src/test/Layout01_test.js [hunk 2]
```diff
@@ -2,26 +2,18 @@ import React from 'react';
...
-describe('Layout01', function() {
-  it('renders the tag', function() {
-    const renderer = TestUtils.createRenderer();
-    renderer.render(<Layout01 />);
-    expect(renderer.getRenderOutput().type).toBe('div');
+describe('Layout01', () => {
+  const wrapper = shallow(<Layout01 />);
+
+  it('verify the no of "div" tags ', () => {
+    expect(wrapper.find('div').length).toEqual(2);
```
Comment (edgibbs) [inline src/test/Layout01_test.js]: Can probably be removed.

File: src/test/NavLink_test.js [hunk 3]
```diff
@@ -13,187 +15,188 @@ describe('NavLink', function() {
...
+      expect(spanElement.props().style).toEqual({ marginRight: '6rem' });
     });
 
-    it('does not have pre & post icon elements', function() {
-      const iElements = ReactTestUtils.scryRenderedDOMComponentsWithTag(
-        inactiveBasicNavLink,
-        'i'
-      );
+    it('does not have pre & post icon elements', () => {
+      const iElements = inactiveBasicNavLink.find('i');
```
Comment (edgibbs) [inline src/test/NavLink_test.js]: So maybe inactiveElements?

File: src/test/NavLink_test.js [hunk 4]
```diff
@@ -13,187 +15,188 @@ describe('NavLink', function() {
...
-    const navLinkWithPreIcon = ReactTestUtils.renderIntoDocument(
+  describe('navlink with pre icon property', () => {
+    const navLinkWithPreIcon = shallow(
       <NavLink href={navLinkHref} text={navLinkText} preIcon={preIconClass} />
     );
-    const liElement = ReactTestUtils.findRenderedDOMComponentWithTag(
-      navLinkWithPreIcon,
-      'li'
-    );
+    const inst = navLinkWithPreIcon.instance();
```
Comment (edgibbs) [inline src/test/NavLink_test.js]: instance or just call .instance() in the expect blocks where it's used

File: src/test/NavLink_test.js [hunk 5]
```diff
@@ -13,187 +15,188 @@ describe('NavLink', function() {
...
-    const navLinkWithPostIcon = ReactTestUtils.renderIntoDocument(
+  describe('navlink with post icon property', () => {
+    const navLinkWithPostIcon = shallow(
       <NavLink href={navLinkHref} text={navLinkText} postIcon={postIconClass} />
     );
-    const liElement = ReactTestUtils.findRenderedDOMComponentWithTag(
-      navLinkWithPostIcon,
-      'li'
-    );
+    const inst = navLinkWithPostIcon.instance();
```
Comment (edgibbs) [inline src/test/NavLink_test.js]: Same full variable name or just use .instance() when needed.

File: src/test/NavLink_test.js [hunk 6]
```diff
@@ -13,187 +15,188 @@ describe('NavLink', function() {
...
         text={navLinkText}
         preIcon={preIconClass}
         postIcon={postIconClass}
       />
     );
-    const liElement = ReactTestUtils.findRenderedDOMComponentWithTag(
-      navLinkWithPrePostIcons,
-      'li'
-    );
+    const inst = navLinkWithPrePostIcons.instance();
```
Comment (edgibbs) [inline src/test/NavLink_test.js]: same comment from above

File: src/test/NavLink_test.js [hunk 7]
```diff
@@ -13,187 +15,188 @@ describe('NavLink', function() {
...
+    const navLinkWithChildren = shallow(
       <NavLink href={navLinkHref} text={navLinkText}>
         <div>Test Child</div>
       </NavLink>
     );
-    const liElement = ReactTestUtils.findRenderedDOMComponentWithTag(
-      navLinkWithChildren,
-      'li'
-    );
+    const inst = navLinkWithChildren.instance();
```
Comment (edgibbs) [inline src/test/NavLink_test.js]: instance or just use of .instance() when needed

File: src/test/PageHeader_test.js [hunk 8]
```diff
@@ -1,71 +1,56 @@
...
+    expect(divElement1.props().className).toBe('page-title text-left');
+    expect(divElement1.props().children).toEqual('CaseName');
+    expect(buttonElement1.props().className).toBe('primary-btn pull-right');
+    expect(buttonElement1.props().children).toEqual('Save Form');
   });
 
-  it('check default props', function() {
-    expect(pageHeader.props.pageTitle).toEqual('CaseName');
+  it('check default props', () => {
+    const inst = pageHeader.instance();
```
Comment (edgibbs) [inline src/test/PageHeader_test.js]: At least instance or pageHeaderInstance for the variable name or just call .instance() when used.

**Comment by edgibbs** [comment]
retest this please

**Comment by RamuKammagani-osi** [comment]
@edgibbs  
Fixed all the variable names and deleted some test cases which doesn't test the behaviour

File: src/test/Alert_test.js [hunk 9]
```diff
@@ -1,17 +1,83 @@
...
+        .at(5)
+        .props().className
+    ).toEqual('alert-cross');
+    expect(
+      comp
+        .find('i')
+        .at(1)
+        .props().className
+    ).toEqual('fa fa-times');
+    const inst = wrapper.instance();
```
Comment (edgibbs) [inline src/test/Alert_test.js]: Still has inst for the variable name.

File: karma.conf.js [hunk 10]
```diff
@@ -39,11 +39,12 @@ var config = function(config) {
       'karma-webpack',
       'karma-jasmine',
       'karma-sourcemap-loader',
+      'karma-mocha-reporter',
     ],
```
Comment (edgibbs) [inline karma.conf.js]: Good call on swapping in mocha reporter, looks much nicer 👍

**Comment by edgibbs** [comment]
retest this please

**Comment by edgibbs** [comment]
Build triggered. sha1 is merged.

**Comment by edgibbs** [comment]
Build started sha1 is merged.

**Comment by edgibbs** [comment]
Everything is green!
 No test results found.

**Pull request merged**